fe/fe-core/src/main/java/com/starrocks/sql/analyzer/SQLLabelBuilder.java

```diff
@@ -0,0 +1,522 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+package com.starrocks.sql.analyzer;
+
+import com.google.common.base.Joiner;
+import com.starrocks.analysis.AnalyticExpr;
+import com.starrocks.analysis.AnalyticWindow;
+import com.starrocks.analysis.ArithmeticExpr;
+import com.starrocks.analysis.ArrayElementExpr;
+import com.starrocks.analysis.ArrayExpr;
+import com.starrocks.analysis.ArrowExpr;
+import com.starrocks.analysis.BetweenPredicate;
+import com.starrocks.analysis.BinaryPredicate;
+import com.starrocks.analysis.CaseExpr;
+import com.starrocks.analysis.CastExpr;
+import com.starrocks.analysis.CompoundPredicate;
+import com.starrocks.analysis.DefaultValueExpr;
+import com.starrocks.analysis.ExistsPredicate;
+import com.starrocks.analysis.Expr;
+import com.starrocks.analysis.FunctionCallExpr;
+import com.starrocks.analysis.FunctionParams;
+import com.starrocks.analysis.GroupingFunctionCallExpr;
+import com.starrocks.analysis.InPredicate;
+import com.starrocks.analysis.InformationFunction;
+import com.starrocks.analysis.IsNullPredicate;
+import com.starrocks.analysis.LikePredicate;
+import com.starrocks.analysis.LimitElement;
+import com.starrocks.analysis.LiteralExpr;
+import com.starrocks.analysis.OrderByElement;
+import com.starrocks.analysis.ParseNode;
+import com.starrocks.analysis.SelectList;
+import com.starrocks.analysis.SelectListItem;
+import com.starrocks.analysis.SlotRef;
+import com.starrocks.analysis.Subquery;
+import com.starrocks.analysis.SysVariableDesc;
+import com.starrocks.analysis.TimestampArithmeticExpr;
+import com.starrocks.sql.ast.AstVisitor;
+import com.starrocks.sql.ast.CTERelation;
+import com.starrocks.sql.ast.ExceptRelation;
+import com.starrocks.sql.ast.FieldReference;
+import com.starrocks.sql.ast.IntersectRelation;
+import com.starrocks.sql.ast.JoinRelation;
+import com.starrocks.sql.ast.QueryRelation;
+import com.starrocks.sql.ast.QueryStatement;
+import com.starrocks.sql.ast.Relation;
+import com.starrocks.sql.ast.SelectRelation;
+import com.starrocks.sql.ast.SetOperationRelation;
+import com.starrocks.sql.ast.SubqueryRelation;
+import com.starrocks.sql.ast.TableFunctionRelation;
+import com.starrocks.sql.ast.TableRelation;
+import com.starrocks.sql.ast.UnionRelation;
+import com.starrocks.sql.ast.ValuesRelation;
+import com.starrocks.sql.optimizer.base.SetQualifier;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.toList;
+
+public class SQLLabelBuilder {
+    public static String toSQL(ParseNode expr) {
```

Comment:
What is it called SQLLabelBuilder? All the functionality is translate `ParseNode` to sql string?